api/src/main/java/org/pmiops/workbench/firecloud/FireCloudService.java

```diff
@@ -82,12 +82,16 @@ WorkspaceACLUpdateResponseList updateWorkspaceACL(String projectName, String wor
 
   void removeUserFromGroup(String email, String groupName);
 
-  boolean isUserMemberOfGroup(String groupName);
+  boolean isUserMemberOfGroup(String email, String groupName);
```

Comment:
Note: the change here (*and in FireCloudServiceImpl and fireCloud.yaml) is a workaround / bugfix to allow the UserService.updateDataAccessLevel method to work correctly within the context of a Cron job, which is updating user data for more than just the currently-authenticated user.

More gory details:

The updateDataAccessLevel method needs to determine if a given user is a member of the registered-tier access group. To do this, we previously made a call like:

fireCloudService.isUserMemberOfGroup(registeredTierGroup);

and the fireCloudService would make a call to FireCloud using the *currently-authenticated user* credentials.

This works fine when handling end-user web requests. But when we're running a batch job, userService.updateDataAccessLevel may be called with a user that *isn't* currently-authenticated.

To fix this, I added the FireCloud API call that retrieves all members of a given group. A comment left by Dan Rodney in FireCloudServiceImpl had said that "There is no endpoint in FireCloud for checking whether a user is a member of a particular group". I can't tell whether Dan didn't recognize the GET method, or was concerned that fetching all users from the registered-tier group would be inefficient – which it will be, but I don't think we need to be too concerned about that for now. Even a list of ~10k emails won't be a huge issue, and we can lean on the FireCloud team to create a more focused API endpoint if we ever need it.

The call can now correctly propagate the target user email from UserService --> FireCloudService:

fireCloudService.isUserMemberOfGroup(userEmail, registeredTierGroup);

(Note: by removing this use of the "endUserGroups" API, we can also follow-up on this by cleaning up the awkward use of two API clients within FireCloudService.)